application/cms/views.py

```diff
@@ -1114,6 +1114,32 @@ def view_measure_version_by_measure_version_id(measure_version_id):
     )
 
 
+@cms_blueprint.route("/<topic_slug>/<subtopic_slug>/<measure_slug>/<version>/edit/data-sources", methods=["GET"])
+@login_required
+@user_has_access
+def search_data_sources(topic_slug, subtopic_slug, measure_slug, version):
+    topic, subtopic, measure, measure_version = page_service.get_measure_version_hierarchy(
+        topic_slug, subtopic_slug, measure_slug, version
+    )
+
+    q = request.args.get("q", "")
+
+    if q:
+        data_sources = DataSource.search(q, limit=100)
```

Comment:
I might be tempted to just drop off the `limit=100` from here, as that's already the default. Then if we decide to change the default later, we don't need to remember to update two places.

Reply:
I sort-of felt that it should be the view's responsibility to say what the limit was, rather than relying on a default set in the model method? Possibly in that case it should be a required argument though - or default to no limit?

Reply:
Makes sense - maybe make it a required parameter.

Reply:
I've made it a optional parameter in 88a67fd50e208dc63967c377b56d536d00c01634 and removed the limit from the SQL if not specified. That felt like the right thing to do, on the grounds that it'd mean we _could_ use that method in places with no limit (eg management commands or admin interfaces)?